File: frontend/src/app/core/services/message.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Message {
  room_id: number;
  message_id: string;
  user_id: number;
  content: string;
  created_at: string;
}

@Injectable({
  providedIn: 'root'
})
export class MessageService {
  private readonly API_URL = 'http://127.0.0.1:5000/rooms';

  constructor(private http: HttpClient) {}

  getMessages(roomId: number): Observable<Message[]> {
    return this.http.get<Message[]>(
      `${this.API_URL}/${roomId}/messages`
    );
  }
}

File: backend/src/api/message.py
from flask import Blueprint, jsonify
from src.services.message_service import MessageService

message_bp = Blueprint(
    "message",
    __name__,
    url_prefix="/rooms"
)


@message_bp.route("/<int:room_id>/messages", methods=["GET"])
def get_messages(room_id):

    messages = MessageService.get_room_messages(room_id)

    return jsonify([
        {
            "room_id": message.room_id,
            "message_id": str(message.message_id),
            "user_id": message.user_id,
            "content": message.content,
            "created_at": message.created_at.isoformat()
        }
        for message in messages
    ]), 200